test/functorch/test_aotdispatch.py

```diff
@@ -2854,27 +2854,109 @@ def forward(self, arg0_1, arg1_1):
 
     # TODO(tmanlaibaatar) properly support functionalizing HOO in
     # predispatch tracing mode
-    @unittest.expectedFailure
     def test_aot_export_predispatch_with_cond(self):
         class M(torch.nn.Module):
             def __init__(self):
                 super().__init__()
-                self.register_buffer("buffer", torch.randn(4, 4))
 
             def forward(self, x):
                 def true_fn(x):
-                    self.buffer.add_(5)
-                    return x.cos() + self.buffer.sum()
+                    y = x
```

Comment:
We should probably add tests for the other higher order ops aside from cond.

Another one that we should probably test in particular is `autograd.Function` being captured properly (since dynamo will lift it into a higher order op)